pkg/kernel/networkservice/inject/common.go

```diff
@@ -61,16 +79,57 @@ func move(logger log.Logger, conn *networkservice.Connection, isMoveBack bool) e
 	}
 	defer func() { _ = targetNetNS.Close() }()
 
+	vfConfig := vfconfig.Config(ctx)
 	ifName := mech.GetInterfaceName()
 	if !isMoveBack {
-		err = moveInterfaceToAnotherNamespace(ifName, curNetNS, curNetNS, targetNetNS)
+		err = moveToContNetNS(vfConfig, ifName, curNetNS, targetNetNS)
 	} else {
-		err = moveInterfaceToAnotherNamespace(ifName, curNetNS, targetNetNS, curNetNS)
+		err = moveToHostNetNS(vfConfig, ifName, curNetNS, targetNetNS)
 	}
 	if err != nil {
-		logger.Warnf("failed to move network interface %s into the target namespace for connection %s", ifName, conn.GetId())
+		// link may not be available at this stage (might be deleted in previous chain element itself) for veth case
+		if strings.Contains(err.Error(), "Link not found") {
+			return nil
+		}
 		return err
 	}
-	logger.Infof("moved network interface %s into the target namespace for connection %s", ifName, conn.GetId())
 	return nil
 }
+
+func moveToContNetNS(vfConfig *vfconfig.VFConfig, ifName string, curNetNS, targetNetNS netns.NsHandle) (err error) {
+	link, _ := kernellink.FindHostDevice("", ifName, targetNetNS)
+	if link != nil {
+		return
+	}
+	if vfConfig != nil && vfConfig.VFInterfaceName != ifName {
+		err = moveInterfaceToAnotherNamespace(vfConfig.VFInterfaceName, curNetNS, curNetNS, targetNetNS)
+		if err == nil {
+			err = renameInterface(vfConfig.VFInterfaceName, ifName, curNetNS, targetNetNS)
```

Comment:
Looks like we need to move interface back in such case.

Reply:
I am afraid that this operation is not idempotent, because after the rename we still have different `vfconfig.VFInterfaceName` and `ifName`, but interface is no more accessible by `vfconfig.VFInterfaceName` because it was renamed.

Reply:
yes, that's why there is a check at line 100-103 to see ifName is already present in contNetNS and if interface is already present, it's returned immediately without an error.